code/code/sys/process.cc

```diff
@@ -184,9 +183,7 @@ void TScheduler::runChar(int pulseNum)
 
   count=max((int)((float)mobCount/11.5), 1);
 
-  for (; tmp_ch; tmp_ch = temp) {
-    temp = tmp_ch->next;  // just for safety
-
+  while (tmp_ch) {
```

Comment:
I see tmp_ch is a member of the TScheduler class but I don't understand why that would be required. In this function it seems preferable to iterate over the entire list each time rather than start in the middle if tmp_ch is pointing to something other than null. Any idea why it's done this way rather than declaring a new locally scoped var pointing to character_list and iterating over the entire list each time?

Reply:
We iterate over 1/11th of the list (see the count-- and break).
If we were to suddenly start iterating over the whole list,
1. CPU consumption would go up
2. lag spikes might appear
3. mob specs would trigger 11x more often.

Reply:
I missed the count logic. So we want to do part of the list and save where we are for next time. Got it.